vendor/github.com/jacobsa/gcloud/gcs/bucket.go

```diff
@@ -210,13 +218,97 @@ func (b *bucket) ListObjects(
 	if listing, err = toListing(rawListing); err != nil {
 		return
 	}
+	fmt.Println((*listing))
+	return
+}
+
+func ListObjectsSCL(
+	ctx context.Context,
+	req *ListObjectsRequest, bucketName string, storageClient *storage.Client) (listing *Listing, err error) {
+	// If client is "nil", it means that there was some problem in initializing client in newBucket function of bucket.go file.
+	var list Listing
+	if storageClient == nil {
+		err = fmt.Errorf("Error in creating client through Go Storage Library.")
+		return
+	}
+
+	query := &storage.Query{
+		Delimiter:                req.Delimiter,
+		Prefix:                   req.Prefix,
+		Projection:               storage.Projection(req.ProjectionVal),
+		IncludeTrailingDelimiter: req.IncludeTrailingDelimiter,
+		//MaxResults: , (Field not present in storage.Query but present in Jacobsa query.)
```

Comment:
IS https://screenshot.googleplex.com/9YB8XbbLVRrpQfp not the same field?

Reply:
But it's not present in the Go storage library's query struct. So cant pass it as a parameter here.
[link](https://pkg.go.dev/cloud.google.com/go/storage#Query)

Reply:
Why are you using a query operation and not a listobject ?

Reply:
because in this [link](https://pkg.go.dev/cloud.google.com/go/storage#BucketHandle.Objects) we have to pass a query parameter.